File: java/backOffice/src/main/java/model/Usuario.java
package model;

import java.time.LocalDateTime;

public class Usuario {

    private int id;
    private String nome;
    private String email;
    private String senha;
    private String perfil;
    private String status;
    private LocalDateTime createdAt;

    public Usuario() {
        this.createdAt = LocalDateTime.now();
        this.perfil    = "operador";
        this.status    = "ativo";
    }

    public static Usuario reconstituir(int id, String nome, String email, String senha,
                                       String perfil, String status, LocalDateTime createdAt) {
        Usuario u   = new Usuario();
        u.id        = id;
        u.nome      = nome;
        u.email     = email;
        u.senha     = senha;
        u.perfil    = perfil;
        u.status    = status;
        u.createdAt = createdAt;
        return u;
    }

    public int getId()                  { return id; }
    public String getNome()             { return nome; }
    public String getEmail()            { return email; }
    public String getSenha()            { return senha; }
    public String getPerfil()           { return perfil; }
    public String getStatus()           { return status; }
    public LocalDateTime getCreatedAt() { return createdAt; }

    public void setId(int id)           { this.id     = id; }
    public void setNome(String nome)    { this.nome   = nome; }
    public void setEmail(String email)  { this.email  = email; }
    public void setSenha(String senha)  { this.senha  = senha; }
    public void setPerfil(String perfil){ this.perfil = perfil; }
    public void setStatus(String status){ this.status = status; }

    @Override
    public String toString() {
        return String.format(
                "Usuario{id=%d, nome='%s', email='%s', perfil='%s', status='%s'}",
                id, nome, email, perfil, status);
    }
}

File: api/src/models/Usuario.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn } from "typeorm"

@Entity("usuario")
export class Usuario {

  @PrimaryGeneratedColumn()
  id: number

  @Column({ type: "varchar", length: 255 })
  nome: string

  @Column({ type: "varchar", length: 255, unique: true })
  email: string

  @Column({ type: "varchar", length: 255 })
  senha: string

  @Column({ type: "varchar", length: 50, default: "operador" })
  perfil: string

  @Column({ type: "varchar", length: 20, default: "ativo" })
  status: string

  @CreateDateColumn()
  created_at: Date
}